### DISCUSSION TITLE
Add ROS API and Sensor Support in the Future?

### DISCUSSION THREAD

**Discussion by liuxiao916** [Genesis-Embodied-AI/genesis-world#45] [Q&A]
Hello,

1. Do you have any plans to support ROS API in the future? For example, I would like to place a camera on a robot and read its data via ROS topic.
2. If a ROS API is not planned, that's fine—I can implement it myself. In that case, could you let me know if it’s possible to attach a fixed camera to the top of a quadruped robot and have it move with the robot?
3. Are there any future plans to support sensors like LiDAR?

Thank you for your time and great work on Genesis!🥰🥰🥰

**Comment by hyp1998** [comment]
等待中.......

**Comment by zhouxian** [comment]
THank you for asking (somehow i remember I have replied to this but turns out i didnt...

Adding support for ROS is definitely helpful, but in the coming weeks we probably don't have manpower to support that...
For attaching camera, we will add an API soon, but meanwhile you could just manually use `camera.update_pose()` to let it move with the link/geom you want.
Yes LiDAR is a must have, and we will add it :)

Regarding ROS, it seems to me that it could be implemented as a wrapper package on top of Genesis, as opposed to from inside genesis. Could you help this? Thank you :)

**Reply by hyp1998** [reply]
I would like to know when LiDAR will be supported?

**Reply by hankun11** [reply]
> THank you for asking (somehow i remember I have replied to this but turns out i didnt...
> 
> Adding support for ROS is definitely helpful, but in the coming weeks we probably don't have manpower to support that... For attaching camera, we will add an API soon, but meanwhile you could just manually use `camera.update_pose()` to let it move with the link/geom you want. Yes LiDAR is a must have, and we will add it :)
> 
> Regarding ROS, it seems to me that it could be implemented as a wrapper package on top of Genesis, as opposed to from inside genesis. Could you help this? Thank you :)

This is great! Do you have a timeline for Lidar support?
